src/js/components/FormField/FormField.js

```diff
@@ -451,7 +458,17 @@ const FormField = forwardRef(
         {...outerProps}
         style={outerStyle}
         onFocus={(event) => {
-          setFocus(shouldKeepFocus());
+          setFocus(containsFocus(formFieldRef.current));
+          if (isDateInputComponent) {
```

Comment:
Can we add a comment around what this block is doing? If i'm following correctly, we are removing formfield focus from the formfield when the dateinput calendar button is in focus?

Also wondering if we should structure as
```
if (!isDateInputComponent) setFocus(containsFocus...)
else {
  const formDescendants
  ...
}